bin/cheribsdtest/cheribsdtest_fault.c

```diff
@@ -131,10 +129,19 @@ CHERIBSDTEST(illegal_perm_seal,
 		cheribsdtest_failure_err("sysctlbyname(security.cheri.sealcap)");
 	sealcap = cheri_andperm(sealcap, ~CHERI_PERM_SEAL);
 	sealed = cheri_seal(ip, sealcap);
-#if !CHERI_SEAL_VIOLATION_EXCEPTION
-	if (!cheri_gettag(sealed))
-		cheribsdtest_success();
+	/* cheri_seal() should tag-clear on failure on all architectures. */
+	if (!cheri_gettag(sealed)) {
+#if CHERI_SEAL_VIOLATION_EXCEPTION
+		/*
+		 * For the transition period from trapping to tag-clearing
+		 * semantics for CHERI-RISC-V, we report a SIGPROT here for CI.
+		 * TODO: remove this once we require tag-clearing.
+		 */
+		if (csr_read(uccsr) & SCCSR_TAG_CLEARING)
```

Comment:
This is quite ugly, but it's temporary and I didn't feel like trying to add another .ct_maybe_traps_or_reports_success = true since that didn't fit nicely with the control flow in cheribsdtest.c